autowheelbarrow.lua

```diff
@@ -0,0 +1,120 @@
+                                    #food_settings.liquid_misc > 0
+                    --dfhack.println("Skipping stockpile due to food items" .. tostring(building))
+
+            end
+
+            -- Check all the flags and skip if any of them are true (for individual stockpiles)
+            if stockpile_settings and stockpile_settings.flags then
+                local flags = stockpile_settings.flags
+                if flags.animals == true or flags.food == true or flags.furniture == true or
+                   flags.corpses == true or flags.refuse == true or flags.ammo == true or flags.coins == true or
+                   flags.bars_blocks == true or flags.gems == true or flags.finished_goods == true or
+                   flags.leather == true or flags.cloth == true or flags.wood == true or
+                   flags.weapons == true or flags.armor == true or flags.sheet == true or
+                   flags[17] == true or flags[18] == true or flags[19] == true or flags[20] == true or
+                   flags[21] == true or flags[22] == true or flags[23] == true or flags[24] == true or
+                   flags[25] == true or flags[26] == true or flags[27] == true or flags[28] == true or
+                   flags[29] == true or flags[30] == true or flags[31] == true then
+                    skip_wheelbarrows = true
+                    --dfhack.println("individual stockpiles Skipping stockpile due to one or more flags being true at " .. tostring(building))
+                end
+            end
+
+            -- Check if all flags are set to false (for the none stockpile)
+            if stockpile_settings and stockpile_settings.flags then
+                local flags = stockpile_settings.flags
+                if flags.animals == false and flags.food == false and flags.furniture == false and
+                   flags.corpses == false and flags.refuse == false and flags.stone == false and flags.ammo == false and
+                   flags.coins == false and flags.bars_blocks == false and flags.gems == false and flags.finished_goods == false and
+                   flags.leather == false and flags.cloth == false and flags.wood == false and flags.weapons == false and
+                   flags.armor == false and flags.sheet == false and
+                   flags[17] == false and flags[18] == false and flags[19] == false and flags[20] == false and
+                   flags[21] == false and flags[22] == false and flags[23] == false and flags[24] == false and
+                   flags[25] == false and flags[26] == false and flags[27] == false and flags[28] == false and
+                   flags[29] == false and flags[30] == false and flags[31] == false then
+                    skip_wheelbarrows = true
+                    --dfhack.println("Skipping NONE stockpile" .. tostring(building))
+                end
+            end
```

Comment:
You got me there. idk what they are for, but they exist so I added them.

Reply:
they only "exist" in the sense that the flag word has 32 bits. they don't have a (known) use and so predicating any behavior on their value will result in unpredictable behavior because their value is unpredictable